mountpoint-s3/src/fstab.rs

```diff
@@ -0,0 +1,227 @@
+use anyhow::anyhow;
+use clap::Parser;
+use mountpoint_s3_fs::cli::CliArgs;
+use std::env;
+use std::ffi::OsString;
+
+#[derive(Parser, Debug)]
+pub struct FsTabCliArgs {
+    pub bucket_name: String,
+    #[clap(value_name = "DIRECTORY")]
+    pub mount_point: String,
+    #[clap(short = 'o')]
+    pub options: String,
+}
+
+impl FsTabCliArgs {
+    pub fn try_to_cli_args(self) -> anyhow::Result<CliArgs> {
+        let cli_arg_list = self.build_cli_arg_list()?;
+        Self::build_cli_args(cli_arg_list)
+    }
+
+    /// Parse the args we've been given into an iterator of options to pass to the main CliArgs parser
+    /// Filters out arguments that aren't meant for Mountpoint, and add `--` to any argument that's not prefixed with `-`.
+    fn build_cli_arg_list(&self) -> anyhow::Result<impl Iterator<Item = OsString>> {
+        let mut options: Vec<String> = Self::split_commas(&self.options)?
+            .into_iter()
+            .filter(|option| !Self::filter_option(option))
+            .map(|option| {
+                if option.starts_with("-") {
+                    option
+                } else {
+                    format!("--{}", option)
+                }
+            })
+            .map(Self::rename_option)
+            .collect();
+
+        Self::validate_options(&options)?;
+        options.retain(|x| x != "--read-write");
+
+        let cli_arg_list = [
+            env::args_os().nth(0).unwrap_or("mount-s3".into()),
+            (&self.bucket_name).into(),
+            (&self.mount_point).into(),
+        ]
+        .into_iter()
+        .chain(options.into_iter().map(|s| s.into()));
+
+        Ok(cli_arg_list)
+    }
+
+    fn build_cli_args(cli_arg_list: impl Iterator<Item = OsString>) -> anyhow::Result<CliArgs> {
+        let mut cli_args = CliArgs::try_parse_from(cli_arg_list).map_err(|e| anyhow!(e))?;
```

Comment:
I feel like the parsing logic should be: parse options into `FsTabCliArgs` and convert `FsTabCliArgs` into `CliArgs` rather than converting `FsTabCliArgs` back to a string and then trying to parse `CliArgs` from there.